src/coprocessor/dag/expr/builtin_miscellaneous.rs

```diff
@@ -28,6 +28,48 @@ impl ScalarFunc {
         }
     }
 
+    pub fn is_ipv4_compat<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<i64>> {
+        let input = try_opt!(self.children[0].eval_string(ctx, row));
+        // Not an IPv6 address, return 0
+        if input.len() != 16 {
+            return Ok(Some(0));
+        }
+
+        let input_bytes: &[u8; 16] = input.as_ref().try_into().unwrap();
+        let prefix_compat = vec![
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        ];
```

Comment:
As I mentioned in https://github.com/tikv/tikv/pull/4405#pullrequestreview-216529973 you don't need to create a Vec.

Reply:
thx for review, fixed